src/auxiliary.jl

```diff
@@ -167,17 +162,22 @@ for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
 end
 
 
-"""
-    max_order(a)
-
-Returns the maximum order of a `Taylor1` or `TaylorN`.
-
-For `a::Taylor1` returns `a.order` while for `a::TaylorN` returns
-`get_order()`.
-"""
-max_order(a::Taylor1) = a.order
+## fixorder ##
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        function fixorder(a::$T, b::$T)
+            a.order == b.order && return a, b
+            a.order < b.order && return $T(copy(a.coeffs), b.order),
```

Comment:
I included the `copy` (knowing about the penalty that this entails) because without it you can change the coefficients of `a` (or `b`) and affect the *original* `a` and `b`, i.e., produce a side effect. 

Probably if `a.order == b.order` should also `return $T(copy(a.coeffs), a.order), $T(copy(b.coeffs), b.order)`, but for the moment, that is where `fixorder` is used, there are no side effects. Should I write that case with the `copy`s, since we are filtering the case `a.order == b.order` ? (You commented on this previously.)